app/src/main/java/com/github/factotum_sdp/factotum/ui/directory/ContactDetailsFragment.kt

```diff
@@ -23,22 +24,40 @@ class ContactDetailsFragment : Fragment() {
         val button = mainView.findViewById<Button>(R.id.button) // connect the button to the layout
         button.setOnClickListener { view ->
             view.findNavController().navigate(R.id.action_contactDetailsFragment2_to_directoryFragment)} // go back to the list of contacts when the button is clicked
```

Comment:
You should encapsulate the logic in the onCreateView function into smaller and add comments.